minicriu-client.c

```diff
@@ -122,6 +124,13 @@ int minicriu_dump(void) {
 	struct savedctx ctx;
 	SAVE_CTX(ctx);
 
+	struct sigaction new;
+	new.sa_handler = mc_sighnd;
+	if (sigaction(MC_THREAD_SIG, &new, NULL)) {
```

Comment:
Here we override MC_THREAD_SIG handler, if any, but don't restore the original one. Does it make sense to move signal-preserving code from line 150 before this point? So after we set original signal handlers back on line 170, we have no traces of minicriu_dump at all.